lib/data/repository/movie_repository.dart

```diff
@@ -1,15 +1,29 @@
-import 'dart:convert';
-import 'package:flutter/services.dart';
+import '../../domain/entity/movie.dart';
 import '../../domain/repository/movie_repository_interface.dart';
 
-import '../../domain/entity/movie.dart';
+import '../datasource/remote/api_service.dart';
+import '../model/data_state.dart';
 
 class MovieRepositoryImp implements IMovieRepository {
+  ApiService apiService = ApiService();
+
+  @override
+  Future<DataState<List<Movie>>> getPopularMovies() {
+    return apiService.getPopularMovies();
+  }
+
+  @override
+  Future<DataState<List<Movie>>> getTopRatedMovies() {
+    return apiService.getTopRatedMovies();
+  }
+
+  @override
+  Future<DataState<List<Movie>>> getNowPlayingMovies() {
+    return apiService.getNowPlayingMovies();
+  }
+
   @override
-  Future<List<Movie>> getMovies() async {
-    final String response =
-        await rootBundle.loadString('assets/mock_data/movies.json');
-    List<dynamic> data = List.from(json.decode(response));
-    return data.map((movie) => Movie.fromJson(movie)).toList();
+  Future<DataState<List<Movie>>> getUpcomingMovies() {
+    return apiService.getUpcomingMovies();
   }
```

Comment:
If you made the change that i mentioned earlier you only need to have one method here like:
```
@override
Future<DataState<List<Movie>>> getMovies({required Endpoint endpoint) {
     switch(endpoint) {
          case Endpoint.nowPlaying: 
               return apiService.getMovies('..../nowPlaying')
     }
}
```
Endpoint is a enum that has all the possible api service call that you have `(nowPlaying, popular, topRated, upcoming)`